pkg/lease/manager.go

```diff
@@ -144,13 +153,20 @@ func (l *manager) createLease(ctx context.Context, obj client.Object, duration t
 func (l *manager) requestLease(ctx context.Context, obj client.Object, leaseDuration time.Duration) error {
 	log.Info("request lease")
 	lease, err := l.getLease(ctx, obj)
-	//couldn't get the lease try to create one
+
 	if err != nil {
-		if err = l.createLease(ctx, obj, leaseDuration); err != nil {
-			l.log.Error(err, "couldn't create lease")
-			return err
+		//couldn't get the lease try to create one
+		if apierrors.IsNotFound(err) {
+			if err = l.createLease(ctx, obj, leaseDuration); err != nil {
+				l.log.Error(err, "couldn't create lease")
+				return err
+			} else {
```

Comment:
I don't think the examples above apply to the pattern which is used here .
i.e:
```
if x {
  return something
} else{
 return somethingElse
}
```

[Here](https://github.com/golang/net/blob/master/http2/write.go#L237-L248) is an example how this exact pattern is used in native go code.

Reply:
> I don't think the examples above apply to the pattern which is used here

What's the difference? 🤔 
The if block always returns, then you don't need the else.

> Here is an example

that's 7 years old...

Reply:
Difference is in readability.
The whole premises of omitting the else is around that:
>This improves the readability of the code by enabling the reader to find the normal path quickly

When both use-cases (if/else) return immediately there is no issue finding the normal path.

The above quote (and examples) makes a point that we should use a return statement promptly, that is why they discourage something like (note the comment about lots of code):
```
// Bad:
if x, err := f(); err != nil {
  // error handling
  return
} else {
  // lots of code that uses x
  // across multiple lines
}
```

in the pattern used here return is prompt and IMO it is arguable what is more readable:
```
//Nested Code ...
  if x {
    return something
  } else{
   return somethingElse
  }
//More Code ...

//Or 
//Nested Code ...
if x {
  return something
}  
return somethingElse
//More Code ... 
```

>that's 7 years old...

That was just the first example I've found while hitting a random pkg, in case a newer example will change your mind I'm willing to try to look for one.

Reply:
> The whole premises of omitting the else is around that:
> 
> > This improves the readability of the code by enabling the reader to find the normal path quickly
> 
> When both use-cases (if/else) return immediately there is no issue finding the normal path.
> 

I don't see why the return statement in the else block makes it more reasonable to have the normal path nested in an unneeded else block 🤷🏼‍♂️ 

> The above quote (and examples) makes a point that we should use a return statement promptly, that is why they discourage something like (note the comment about lots of code):
> 
> ```
> // Bad:
> if x, err := f(); err != nil {
>   // error handling
>   return
> } else {
>   // lots of code that uses x
>   // across multiple lines
> }
> ```
> 

I understand this example as "a var initialized in the if statement excuses an else block when it has a few lines of code only"... which doesn't apply here 

> [...] IMO it is arguable what is more readable:

I agree on readabilty being arguable... and since we obviously disagree, let's agree to disagree 🤷🏼‍♂️